packages/react-native-reanimated/apple/reanimated/apple/REAModule.mm

```diff
@@ -193,9 +193,10 @@ - (void)setBridge:(RCTBridge *)bridge
 
 - (void)setReaSurfacePresenter
 {
-  if (reaSurface == nil) {
-    // we need only one instance because SurfacePresenter is the same during the application lifetime
-    reaSurface = [[REAInitializerRCTFabricSurface alloc] init];
+  if(_surfacePresenter && ![_surfacePresenter surfaceForRootTag:reaSurface.rootTag]) {
```

Comment:
In brownfield scenario, when we create, destroy and then create the module again, `reaSurface is still the same, but `_surfacePresenter` is a new one so we need to add it again.